src/System.Memory/src/System.Memory.csproj

```diff
@@ -0,0 +1,55 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;$(NetFrameworkMinimum);netcoreapp2.1</TargetFrameworks>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <StrongNameKeyId>Open</StrongNameKeyId>
+    <Nullable>disable</Nullable>
+    <IsPackable>false</IsPackable>
+    <VersionPrefix>4.5.5</VersionPrefix>
+    <VersionPrefix Condition="'$(IsPackable)' == 'true'">4.6.0</VersionPrefix>
+    <AssemblyVersion>4.0.1.2</AssemblyVersion><!-- NO-INCREMENT: This version is frozen for .NET Standard and .NETCoreApp because the assembly ships inbox. -->
+    <AssemblyVersion Condition="$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETFramework'">4.0.1.2</AssemblyVersion>
+    <AssemblyVersion Condition="'$(IsPackable)' == 'true' and $([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETFramework'">4.0.2.0</AssemblyVersion>
+    <PackageValidationBaselineVersion>4.5.5</PackageValidationBaselineVersion>
+    <IsPlaceholderTargetFramework Condition="'$(TargetFramework)' == 'netcoreapp2.1'">true</IsPlaceholderTargetFramework>
+    <DefineConstants Condition="'$(IsPlaceholderTargetFramework)' != 'true'">$(DefineConstants);NETSTANDARD</DefineConstants>
```

Comment:
This looks wrong. Why would we specify the NETSTANDARD define for .NET Framework? Note that you don't need to have this define manually specified when targeting .NET Standard. It's an implicit define added by the SDK.

Reply:
Hmmm I'm actually not sure about this. If I invert the condition, the build fails because some types are not found anymore. If I remove the NETSTANDARD constant, it also fails.

Here's how it was defined back in 2.1:

[/src/System.Memory/src/System.Memory.csproj#L8-L9](https://github.com/dotnet/corefx/blob/8b6f40185c25372943ac4a8ab94b860008ad2120/src/System.Memory/src/System.Memory.csproj#L8-L9)
```xml
<IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'netcoreapp' OR '$(TargetGroup)' == 'uap'">true</IsPartialFacadeAssembly>
<DefineConstants Condition="'$(IsPartialFacadeAssembly)' != 'true'">$(DefineConstants);netstandard;FEATURE_PORTABLE_SPAN</DefineConstants>
```

Please ignore the property name `IsPartialFacadeAssembly`. Just look at the property, value, which is very similar to what we have now with `IsPlaceholderTargetFramework`.

Reply:
Based on that it sounds like you don't need that define at all anymore as it previously applied to both the .NET Framework and the .NET Standard build.

Reply:
Agree with Viktor - just remove all those ifdefs.  All builds should now behave as if `netstandard` and `FEATURE_PORTABLE_SPAN` are set.

Reply:
Ah, got it. Will set that code as default and will remove the ifdefs. 👍